crates/m3/src/builder/witness.rs

```diff
@@ -1095,11 +1095,13 @@ impl<'a, F: TowerField, P: PackedField<Scalar = F>> TableWitnessSegment<'a, P> {
 			.columns
 			.iter()
 			.zip(expr_circuit.vars_usage())
-			.enumerate()
-			.map(|(partition_index, (col_id, used))| {
+			.map(|(col_id, used)| {
 				used.then(|| {
-					// TODO: conjuring up Col is discouraged.
-					self.get(Col::<FSub, V>::new(*col_id, ColumnPartitionIndex(partition_index)))
+					let col = self
```

Comment:
Hmm, this makes me realize there's actually a correctness issue with `get` (and `get_mut`), which happens when `upcast_col` is used. If we upcast a `Col<F1, V>` into an extension field column, `Col<F2, V>`, then casting the underlying data as `F2` elements is actually not correct. They need to be read as `F1` data and then explicitly converted... :-/ The only fix I can think of really is to check that the column's tower height matches `FSub::TOWER_LEVEL` in get, and return an error otherwise.

Reply:
We've decided to just leave an assert in this PR and do a proper fix as described below:

https://linear.app/irreducible/issue/CRY-371/extending-eval-expr